projects/plugins/boost/app/modules/image-size-analysis/Image_Size_Analysis_Fixer.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Image_Size_Analysis;
+
+class Image_Size_Analysis_Fixer {
+
+	public static function setup() {
+		add_filter( 'the_content', array( __CLASS__, 'fix_content' ), 99 );
+		add_filter( 'wp_calculate_image_srcset', array( __CLASS__, 'fix_image_attachments' ), 10, 5 );
+		add_filter( 'wp_calculate_image_sizes', array( __CLASS__, 'fix_image_sizes' ), 10, 5 );
+	}
+
+	public static function fix_url( $url ) {
+		$url = preg_replace( '/-\d+x\d+\.jpg/', '.jpg', $url ); // remove "-1024x768" from the end of the URL.
+		$url = preg_replace( '/\?.*/', '', $url ); // remove query string.
+		$url = str_replace( 'i0.wp.com/', '', $url ); // remove i0.wp.com/ from the beginning of the URL.
+		return $url;
+	}
+
+	public static function get_fixes( $post_id ) {
+		static $fixes = array();
+
+		if ( isset( $fixes[ $post_id ] ) ) {
+			return $fixes[ $post_id ];
+		}
+
+		$fixes[ $post_id ] = get_post_meta( $post_id, 'jb_image_fixes', true );
+		if ( ! $fixes[ $post_id ] ) {
+			$fixes[ $post_id ] = array();
+		}
+
+		return $fixes[ $post_id ];
+	}
+
+	// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	public static function fix_image_attachments( $sources, $size_array, $image_src, $image_meta, $attachment_id ) {
```

Comment:
Is `image_src` the `<img src=""` tag or the image URL? This can be confusing because there are functions like [`wp_get_attachment_image_src`](https://developer.wordpress.org/reference/functions/wp_get_attachment_image_src/) that produce HTML. 

If the function does expect HTML, maybe this should be called `image_html` or `image_html_src` ?

Reply:
It's the URL. I'll rename it $image_url.